ldms/src/sampler/ibmad_sampler/ibmad_sampler.c

```diff
@@ -36,13 +36,28 @@
 /* So far I cannot find a header that defines these for us */
 #define PORT_STATE_ACTIVE 4
 
+#define MAX_CA_NAMES 32
+#define PORT_FILTER_NONE 0
+#define PORT_FILTER_INCLUDE 1
+#define PORT_FILTER_EXCLUDE 2
+struct port_name {
+	char ca_name[UMAD_CA_NAME_LEN];
+	uint64_t port_bits;
+	/*< 1: matches $name.*. bits & 1 << $port matches $name.$port.
+	* this works because ib disallows port number 0. */
+};
+#define ALL_PORTS(p) (p & 1)
```

Comment:
This is too unnecessarily clever for its own good. It makes the code harder to read and understand. Just set all of the bits when it should match all of the ports, and this can go way.

Reply:
yep.